### PR TITLE
feat(compile): able to sync folder after compile

### LINKED ISSUE #1209: syncFolder feature
We need the ability to add arbitrary content to the sasjsbuild folder (prior to build).  The content could be mocked files (eg JS) or even SAS programs, for developers who wish to add content outside of the regular compile process.

The implementation will be a single attribute at both ROOT and TARGET level as follows:

`"syncFolder": "project/location/myFolder",`

* Any and all files & subfolders inside this will be added inside the sasjsbuild folder
* Any same-named SAS programs (eg `services/common/appinit.sas`) will be overwritten if they exist in the syncFolder
* The `syncFolder` content(s) should be added _after_ the regular compile process (so will replace the compiled version if there is a conflict)
* The ROOT level `syncFolder` is added first, then the TARGET level (if either exist).  So, TARGET files will overwrite ROOT, which will overwrite COMPILED files.
* `sasjs compile` should work for a minimal `sasjsconfig.json` that contains nothing else, eg:

```json
{
  "$schema": "https://raw.githubusercontent.com/sasjs/utils/main/src/types/sasjsconfig-schema.json",
  "syncFolder": "project/location/myFolder",
  "defaultTarget": "server",
  "targets": [
    {
      "name": "sasjs",
      "serverUrl": "",
      "serverType": "SASJS",
      "syncFolder": "target/specificFolder"
    }
}
```

In this case only the contents of `myFolder`  and `specificFolder` added to the `sasjsbuild` folder. 

Tests should include:

- [ ] the syncFolder-only use case above
- [ ] same named SAS program (should be successfully overwritten)
- [ ] file in a new folder, 3 levels deep

The schema json should also be updated.  

Note - this feature should work on ALL server types (SASJS, SAS9 and SASVIYA)

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Fixes #1209 

## Intent

Should be able to copy files from syncFolder of `Configuration.Root` and then of `Target`.

## Implementation

What code changes have been made to achieve the intent.

## Checks

- [ ] Code is formatted correctly (`npm run lint:fix`).
- [ ] Any new functionality has been unit tested.
- [ ] All unit tests are passing (`npm test`).
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [ ] Reviewer is assigned.

### CODE REVIEW

File: src/commands/compile/compile.ts [hunk 1]
```diff
@@ -378,3 +381,15 @@ async function compileWeb(target: Target) {
       })
   }
 }
+
+const syncFolder = async (target: Target) => {
```
Comment (YuryShkoda) [inline src/commands/compile/compile.ts:385]: In my opinion, this logic should belong to `copySyncFolder` function.

**Comment by saadjutt01** [comment]
Compiling an app, involves process.env.variables, whereas `sasjs/utils` is not introduced to that/ dependent to it.


Our current implement of compile a single file is,
- gathering all the required information and passing it to `loadDependenciesFile` ( sasjs/utils' module )
Which is minimal piece of code from flow of compiling process.

And syncFolder is updating/overwriting entire `sasjsbuild` folder. **It's not part of single file compile.**

We can move `syncFolder` feature to `sasjs/utils`, when we are moving compile process of project.

If we are still motivated to move `copySyncFolder` to `sasjs/utils` in current codebase and again extracting required information(i.e. process env variables) it will be single line function that is `copy(...)`

File: src/commands/compile/compile.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/compile/compile.ts:385]: added separate [comment](https://github.com/sasjs/cli/pull/1214#issuecomment-1158958162) on this PR

File: src/commands/compile/internal/copySyncFolder.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { copy, folderExists, getAbsolutePath } from '@sasjs/utils'
+
+export const copySyncFolder = async (syncFolder: string) => {
+  const { buildSourceFolder, buildDestinationFolder } = process.sasjsConstants
+  const mocksFolder = getAbsolutePath(syncFolder, buildSourceFolder)
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply src/commands/compile/internal/copySyncFolder.ts]: updated

**Review by allanbowe — APPROVED** [review]

**Pull request merged**